src/objects.h

```diff
@@ -235,7 +235,8 @@ class HeapObject : public Object {
 
   static const int CLASS_ID_BIT_SIZE = 10;
   static const int CLASS_ID_OFFSET = CLASS_TAG_OFFSET + CLASS_TAG_BIT_SIZE;
-  static const uword CLASS_ID_MASK = (1 << CLASS_ID_BIT_SIZE) - 1;
+  // This mask lets class_id() return negative values.
+  static const uword CLASS_ID_MASK = -1;
```

Comment:
We use negative class ids for the GC's on-heap pseudo-objects.

Reply:
Maybe add that to the comment?